src/js/services/ng-scores.js

```diff
@@ -328,6 +332,54 @@ define('services/ng-scores',[
         };
 
         /**
+         * Enable automatic refresh of scores if no e.g. manual
+         * refresh happened for a while. This is mostly used as
+         * a backup scenario in case we e.g. missed an (MHub) event,
+         * or when MHub is not available.
+         * Use disableAutoRefresh() when no longer needed (e.g. on
+         * controller destruction).
+         */
+        Scores.prototype.enableAutoRefresh = function() {
+            this._autoRefreshUsers++;
+            this._updateAutoRefreshTimer();
+        };
+
+        /**
+         * Unregister interest in auto-refresh. See enableAutoRefresh().
+         */
+        Scores.prototype.disableAutoRefresh = function () {
+            if (this._autoRefreshUsers <= 0) {
+                throw new Error("enableAutoRefresh()/disableAutoRefresh() calls mismatched");
+            }
+            this._autoRefreshUsers--;
+            if (this._autoRefreshUsers === 0) {
+                // Only stop the timer when needed, otherwise
+                // we unnecessarily restart it
+                this._updateAutoRefreshTimer();
+            }
+        };
+
+        /**
+         * Restart refresh timer, if anyone is interested in auto-refreshing.
+         */
+        Scores.prototype._updateAutoRefreshTimer = function() {
+            var self = this;
+            if (this._autoRefreshInterval !== undefined) {
+                $interval.cancel(this._autoRefreshInterval);
+                this._autoRefreshInterval = undefined;
+            }
+            if (this._autoRefreshUsers > 0) {
+                this._autoRefreshInterval = $interval(function() {
+                    // Reload scores from server (if we're not already
+                    // in the middle of something).
+                    if (self._updating <= 0) {
```

Comment:
Better extract a function isUpdating() that checks the _updating counter, rather than comparing the counter against 0 on several places